src/components/Modal/VPDelegationModal.tsx

```diff
@@ -0,0 +1,124 @@
+import './VPDelegationModal.css'
+
+import Markdown from 'decentraland-gatsby/dist/components/Text/Markdown'
+import Avatar from 'decentraland-gatsby/dist/components/User/Avatar'
+import useFormatMessage, { useIntl } from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import { navigate } from 'decentraland-gatsby/dist/plugins/intl'
+import { Button } from 'decentraland-ui/dist/components/Button/Button'
+import { Close } from 'decentraland-ui/dist/components/Close/Close'
+import { Modal, ModalProps } from 'decentraland-ui/dist/components/Modal/Modal'
+import { Table } from 'decentraland-ui/dist/components/Table/Table'
+import React, { useRef, useMemo, useState } from 'react'
+import { IntlShape } from 'react-intl'
+
+import { snapshotUrl } from '../../entities/Proposal/utils'
+import useVotingPowerBalance from '../../hooks/useVotingPowerBalance'
+import exampleDelegates from '../../modules/delegates/example_delegates.json'
+import locations from '../../modules/locations'
+import Sort from '../Icon/Sort'
+
+type Delegate = {
+  address: string
+  ens_name: string | undefined
+  total_vp: number
+}
+
+type VPDelegationModalProps = Omit<ModalProps, 'children'> & {
+  vp: number
+  space: string
+}
+
+function addressShortener(address: string) {
+  return address.substring(0, 6) + '...' + address.substring(38, 42)
+}
+
+function tableCell(delegate: Delegate, picked_by: number, intl: IntlShape, arrowRef: React.RefObject<HTMLDivElement>) {
+  return (
+    <>
+      <Table.Cell>
+        <span className="Candidate">
+          <Avatar address={delegate.address} size="small" />
+          {delegate.ens_name || addressShortener(delegate.address)}
```

Comment:
There is an `Address` component that already shortens address for us!

With @andyesp we were also thinking of creating a `Username` component that we can reuse, because we are calling it in several places (like [here](https://github.com/andyesp/governance/blob/46c9684c07c8dfb14f3ddea4f8e9a60fc00b0667/src/components/Section/DelegateLabel.tsx#L13))

```typescript
const Username = ({ address }: { address: string }) => {
  const { profile } = useProfile(address)

  return (
    <Link className="DetailsSection__Value" href={locations.balance({ address: address })}>
      {profile && profile.name && <Avatar size="mini" address={profile.ethAddress} style={{ marginRight: '.5rem' }} />}
      {profile && profile.name}
      {(!profile || !profile.name) && !!address && (
        <Blockie scale={3} seed={address || ''}>
          <Address value={address || ''} />
        </Blockie>
      )}
    </Link>
  )
}
```

I'm thinking that you might not even need the ens name in the json because the `useProfile` hook already looks it up for you with the address 🤔 

Maybe we could take this into account and push the refactored component to the delegation branch, and make it so it can receive some classes/styles as props, so we all can use it? What do you say @andyesp?

Reply:
- addressShortener replaced by the Address component
- Use of the hook useProfile to get the name

Maybe the 3 of us could discuss how to refactor this component because it is not clear to me how to make it generic for the different use cases.